lib/charms/layer/openstack.py

```diff
@@ -345,6 +353,15 @@ def __init__(self, app_name, port, subnet, algorithm, fip_net,
         self._try_load_cached_info()
         self._impl = self._get_impl()
 
+    def get_all(self):
+        lbs = []
+        for lb in self._impl.list_loadbalancers():
+            if lb['name'].startswith('openstack-integrator-{}'
+                                     .format(MODEL_SHORT_ID)):
+                lbs.append(lb)
+
+        return lbs
```

Comment:
here's an opportunity to use a nice list-generator
```suggestion
        return [
            lb for lb in self._impl.list_loadbalancers()
            if lb['name'].startswith('openstack-integrator-{}'.format(MODEL_SHORT_ID))
        ]
```

Reply:
I think you'll find that's a comprehension not a generator ;)

Reply:
sleepy-- yep